File: apps/household-hub/client/core/data/src/commonMain/kotlin/com/homelab/household/data/mapper/SpaceDataMapper.kt
package com.homelab.household.data.mapper

import com.homelab.household.data.dto.SpaceReadDto
import com.homelab.household.domain.model.Space
import com.homelab.household.domain.model.SpaceType

object SpaceDataMapper {
    fun toDomain(dto: SpaceReadDto): Space =
        Space(
            id = dto.id,
            name = dto.name,
            type = if (dto.type.equals("household", ignoreCase = true)) SpaceType.HOUSEHOLD else SpaceType.PERSONAL,
            ownerId = dto.owner_id,
            settings = dto.settings,
            createdAt = dto.created_at,
        )

    fun toWireSettings(settings: Map<String, Any?>): Map<String, String> =
        settings.mapValues { it.value?.toString() ?: "" }
}

File: apps/household-hub/backend/app/data/mappers/space_data_mapper.py
from app.domain.entities.space import Space
from app.data.models.space_model import SpaceModel


class SpaceDataMapper:
    @staticmethod
    def to_domain(model: SpaceModel) -> Space:
        return Space(
            id=model.id,
            name=model.name,
            type=model.type,
            owner_id=model.owner_id,
            settings=model.settings or {},
            created_at=model.created_at,
            updated_at=model.updated_at,
        )

    @staticmethod
    def to_model(entity: Space) -> SpaceModel:
        return SpaceModel(
            id=entity.id,
            name=entity.name,
            type=entity.type,
            owner_id=entity.owner_id,
            settings=entity.settings or {},
            created_at=entity.created_at,
            updated_at=entity.updated_at,
        )
